quodlibet/quodlibet/ext/events/auto_update_tags_in_files.py

```diff
@@ -0,0 +1,246 @@
+    _("When a song was played or skipped, the album of that song will be "
+      "checked. If every song in the album has been rated and at least one "
+      "has no ratings or play counts stored in its file, the plugin will "
+      "write the tags to the songs' files. \n\nUse this to avoid constant "
+      "file updates, but be aware that once an album was updated, you'll have "
+      "to use the 'Update Tags in Files' plugin whenever you want modified "
+      "ratings and play counts to be written to the files.")]
+
+PLAY_COUNT_ABOVE_ZERO_TOOLTIP = _(
+        "When the plugin will write the tags of an album, it will "
+        "first set the play count of the songs who are zero to one.\n"
+        "Sometimes you already know that you don't like a song, so "
+        "setting it to one when saving can be useful later on, when "
+        "searching for albums you have fully listened to (%s).")
+
+
+class Config(object):
+    _config = PluginConfig("autoupdatetagsinfiles")
+
+    update_strategy = IntConfProp(_config, "update_strategy",
+                                  UpdateStrategy.AFTER_PLAY_NOT_SKIP)
+    ensure_play_counts_above_zero = BoolConfProp(  # useful for searching
+            _config, "ensure_play_counts_above_zero", False)
+
+
+CONFIG = Config()
+
+
+class AutoUpdateTagsInFiles(EventPlugin):
+    PLUGIN_ID = "AutoUpdateTagsInFiles"
+    PLUGIN_NAME = _("Auto Update Tags in Files")
+    PLUGIN_DESC = _("When songs were played, update the tags in their files. "
+                    "This will ensure play counts and ratings are up to date.")
+    PLUGIN_ICON = Icons.DOCUMENT_SAVE
+
+    def PluginPreferences(self, _):
+        return AutoUpdateTagsPrefs()
+
+    def plugin_on_song_ended(self, song, skipped):
+        if song is None or not is_writable(song):
+            return
+
+        strategy = CONFIG.update_strategy
+        if strategy == UpdateStrategy.AFTER_PLAY_NOT_SKIP:
+            if not skipped:
+                self._try_to_update_song(song)
+            return
+        elif strategy == UpdateStrategy.AFTER_PLAY_OR_SKIP:
+            self._try_to_update_song(song)
+            return
+
+        self._update_album_if_fully_rated(song.album_key)
+
+    def _try_to_update_song(self, song_wrapper):
+        try:
+            song_wrapper._needs_write = True
+            self._write_tags_to_files([song_wrapper])
+        except Exception as e:
+            ErrorMessage(app.window,
+                         SONG_ERROR_FMT.format(song_wrapper, self.PLUGIN_NAME),
+                         "{}\n\n{}".format(e, format_exc())).run()
+
+    def _update_album_if_fully_rated(self, album_key):
+        album = app.library.albums.get(album_key, None)
+        if album is None:
+            return
+
+        songs = album.songs
+        # first check for ratings to avoid costly file checks
+        if not songs or not all(song.has_rating for song in songs):
+            return
+
+        one_or_more_need_update = False
```

Comment:
You're right. I have no idea what happened there. I think I may have had multiple lines (instead of the `has` function) in that loop, moved it and forgot to simplify.

Looks much better now!